static/js/components/tiles/chart_tile.tsx

```diff
@@ -28,29 +30,78 @@ interface ChartTileContainerProp {
   sources: Set<string>;
   children: React.ReactNode;
   replacementStrings: ReplacementStrings;
+  // Whether or not to allow chart embedding action.
+  allowEmbed: boolean;
+  // callback function for getting the chart data as a csv. Only used for
+  // embedding.
+  getDataCsv?: () => string;
   // Extra classes to add to the container.
   className?: string;
 }
 
 export function ChartTileContainer(props: ChartTileContainerProp): JSX.Element {
+  const containerRef = useRef(null);
+  const embedModalElement = useRef<ChartEmbed>(null);
   const title = props.title
     ? formatString(props.title, props.replacementStrings)
     : "";
   return (
     <div
       className={`chart-container ${props.className ? props.className : ""}`}
+      ref={containerRef}
     >
-      <h4>{title}</h4>
+      {title && <h4>{title}</h4>}
       {props.children}
-      {props.sources && (
-        <footer>
+      <footer id="chart-container-footer">
+        {!_.isEmpty(props.sources) && (
           <div className="sources">
             Data from {getSourcesJsx(props.sources)}
           </div>
-        </footer>
-      )}
+        )}
+        <div className="outlinks">
+          {props.allowEmbed && (
+            <a
+              href="#"
+              onClick={(event) => {
+                event.preventDefault();
+                handleEmbed();
+              }}
+            >
+              Export
+            </a>
+          )}
+        </div>
+      </footer>
+      <ChartEmbed ref={embedModalElement} />
```

Comment:
should this also be wrapped with props.allowEmbed?

Reply:
yes, good call, will make a follow up PR for this